test/persist-txn-fencing/mzcompose.py

```diff
@@ -141,98 +141,100 @@ def run_workload(c: Composition, workload: Workload) -> None:
         "mz_second": workload.persist_txn_second,
     }
 
-    with c.override(
-        *[
-            Materialized(
-                name=mz_name,
-                external_cockroach=True,
-                external_minio=True,
-                sanity_restart=False,
-                additional_system_parameter_defaults={
-                    "persist_txn_tables": mzs[mz_name]
-                },
-                # Shadow catalog can't gracefully handle fences.
-                catalog_store="stash",
-            )
-            for mz_name in mzs
-        ]
-    ):
-        c.up("mz_first")
-
-        c.sql(
-            """
-                ALTER SYSTEM SET max_tables = 1000;
-                ALTER SYSTEM SET max_materialized_views = 1000;
-            """,
-            port=6877,
-            user="mz_system",
-            service="mz_first",
-        )
+    # Run the test twice with each catalog implementation.
```

Comment:
Thoughts on taking the catalog impl as a parameter to `run_workload` and moving this loop into `run_default`?

Reply:
That sounds much cleaner, just updated.